server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -321,6 +322,18 @@ public FieldContext fieldContext() {
         return fieldContext;
     }
 
+    /**
+     * If this is a numeric field backed values source type, return the type of the numeric field backing it.
+     * Otherwise return null.
+     */
+    public NumberFieldMapper.NumberType getNumericType() {
```

Comment:
Should this be `getNumberType` instead? I'm always confused by the difference between `NumericType` and `NumberType`.

Or, should this be `double round(double)`? I think, maybe, `NumberType` is sort of a mapping concept. Or, at least, a `NumberFieldMapper` concept. Also, that way maybe it'd be possible to apply the fix to `scaled_float` without forcing it to implement some of the internals of `NumberFieldMapper`.

Reply:
Yeah, it should be `getNumberType`, if we keep it.  You raise good points about why we might not want to keep it though. `NumberType` is a mapping concept, and I don't feel great exposing it here.  Especially since we're weirdly abusing its parse function to make this work.  But I do think we need to delegate this to the Field Type somehow, and this was the most likely candidate I saw for that.

Reply:
We *sort of* already have a method for this - `rangeQuery` and `termQuery`. They do the rounding we want. They just return queries we can't do anything with. I'd be ok with a new method on `MappedFieldType` that could round numeric bounds.

Reply:
It looks to me that `rangeQuery` and `termQuery` just call parse, same as I'm doing here.  I'm hesitant to add this to mapped field type, because it is a fundamentally number-oriented concept.  So you'd have one more method that throws `UnsupportedOperationException` all over the place, except for a few special cases.  I hate throwing `UnsupportedOperationException`; I feel like it breaks the whole idea of polymorphism.    In this case especially, it feels like we're going through a lot of effort to support keeping `ScaledFloatFieldType` from having to extend `NumberFieldType`, which maybe feels like not a great goal?

Having said that, I do think exposing the `NumberType` is heavy handed.  How would you feel about a method on `NumberFieldType` called `double coercePrecision(Object)` or something like that?  It'd just call parse, but the meaning is a little clearer, I think (or at least it gives me a spot to drop javadoc and make the meaning clearer).  It doesn't solve the scaled float case, but I'm not sure how big a deal that is?

Anyway, if you really feel like this belongs on `MappedFieldType`, I can do that too.  WDYT?